### PR TITLE
Draft circleCI config file for testing & building static HTML from notebooks

### LINKED ISSUE #14: think about automated testing the notebooks

### PR DESCRIPTION
**Pull request by adrn**
Here's a CI config file to get the conversation started. 

I pulled out and revamped the notebook build machinery from astropy-tutorials and made a standalone package for executing and converting (to HTML) collections of notebooks: https://github.com/adrn/nbstatic

I decided not to go with @eteq's nbpages because I think this way is simpler: I implemented the build machinery as a package with command entry points. The advantage there is that we don't have to restructure the workshop materials or add any additional files, we just have to add `nbstatic` as a test dependency and run:
```
nbstatic execute .
```
to this circleci config.

I think someone will have to enable circleci for this repo in order to properly test this out!

This fixes #14 

**Note: this has commits from #108 because it doesn't make sense to edit the old coordinates notebooks in here**

### CODE REVIEW

**Comment by pllim** [comment]
Wut? No Windows?

**Comment by adrn** [comment]
Another way you could have asked: "Thanks for taking the lead on this! Have you thought about adding a Windows build as another workflow job?"

**Comment by pllim** [comment]
Thanks for taking the lead on this! Have you thought about adding a Windows build as another workflow job? 😸

**Comment by adrn** [comment]
A strange turn of events: the astropy 4.0 rc install is currently broken for all platforms *except* windows because of an issue with astropy-helpers and setuptools...so now the *only* test I included to actually run is the windows build 😬 

Note that the current set of commits just tests setting up the environment -- on to running the actual notebooks next...

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,7 +50,7 @@ make_env: &make_env
   name: Make workshop environment
   command: |
     conda env create -n astropy-workshop --file 00-Install_and_Setup/environment.yml
-    conda activate astropy-workshop
+    source activate astropy-workshop
```
Comment (pllim) [inline .circleci/config.yml]: Wait... I thought `source activate` is a very old command that should not be used anymore, no?

**Comment by pllim** [comment]
FWIW, @bsipocz fixed the helpers issue.

File: .circleci/config.yml (hunk 1 above)
Comment (adrn) [reply .circleci/config.yml]: Going based on this: https://stackoverflow.com/questions/58849994/running-tests-at-circleci-in-conda-environment

It seems I can either do: 
```
conda init bash
source ~/.bashrc
conda activate astropy-workshop
```
or
``
source activate astropy-workshop
```
🤷‍♂

Comment (pllim) [reply .circleci/config.yml]: Is the conda version on the CircleCI docker image too old or something?

**Comment by adrn** [comment]
[It worked](https://app.circleci.com/jobs/github/astropy/astropy-workshop/31/parallel-runs/0/steps/0-107), finally! 

Two things to do:
- Mark expected failing cells with the "raises-exception" tag (per nbconvert)
- Make nbstatic actually error if it collects any cell exceptions (right now it prints them but doesn't actually raise an exception, so the tests passed incorrectly...)

**Comment by pllim** [comment]
How do I use this "raises-exception" tag?

**Comment by adrn** [comment]
It's poorly documented, but in Jupyter:
View > Cell toolbar > tags

Then above the cell you expect to raise an exception, add the tag: raises-exception

e.g.,
image

image

**Comment by pllim** [comment]
Thanks, @adrn ! TIL.

p.s. Does `TheJoker` raise `WhySoSerious` exception?

File: .circleci/config.yml (hunk 1 above)
Comment (adrn) [reply .circleci/config.yml]: I don't think there is one, but link me to it if I'm wrong!

**Comment by adrn** [comment]
(if you ignore the coordinates_solutions failure...)The build is now failing on notebooks that have real errors (Specutils_analysis) or can't be run automatically (photutils/image_psf_photometry_withNIRCam.ipynb), so I think this is working as intended! The photutils notebook requires some manual downloads - perhaps we should switch those to cells that automatically download the files?

**Comment by pllim** [comment]
@adrn , fixing the notebooks is out of scope, right? Can we merge this?

**Comment by adrn** [comment]
> fixing the notebooks is out of scope, right?

Yes!

> Can we merge this?

This now contains commits from #108, so it depends on whether you want to wait until someone reviews that, or just trust :)

**Comment by pllim** [comment]
Since, you are teaching Coordinates anyway, review is just a formality at this point (ans: trust).

**Pull request merged**